be1-go/hub/organizer_test.go

```diff
@@ -0,0 +1,347 @@
+package hub
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"os"
+	"strconv"
+	"student20_pop"
+	"student20_pop/message"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.dedis.ch/kyber/v3"
+	"go.dedis.ch/kyber/v3/sign/schnorr"
+	"golang.org/x/xerrors"
+)
+
+type keypair struct {
+	public    kyber.Point
+	publicBuf message.PublicKey
+	private   kyber.Scalar
+}
+
+var organizerKeyPair keypair
+
+var suite = student20_pop.Suite
+
+var oHub *organizerHub
+
+var laoCounter = 0
+
+func generateKeyPair() (keypair, error) {
+	secret := suite.Scalar().Pick(suite.RandomStream())
+	point := suite.Point().Pick(suite.RandomStream())
+	point = point.Mul(secret, point)
+
+	pkbuf, err := point.MarshalBinary()
+	if err != nil {
+		return keypair{}, xerrors.Errorf("failed to create keypair: %v", err)
+	}
+	return keypair{point, pkbuf, secret}, nil
+}
+
+func getTime() message.Timestamp {
+	return message.Timestamp(time.Now().Unix())
+}
+
+func createLao(o *organizerHub, oKeypair keypair) (string, *laoChannel, error) {
+	// Data of the Lao
+	name := strconv.Itoa(laoCounter)
+	laoCounter += 1
+	creation := getTime()
+	laoID, err := message.Hash(message.Stringer("L"), oKeypair.publicBuf, creation, message.Stringer(name))
+	if err != nil {
+		return "", nil, err
+	}
+	// Creation of the data
+	data := &message.CreateLAOData{
+		GenericData: &message.GenericData{
+			Action: message.DataAction(message.CreateLaoAction),
+			Object: message.LaoObject,
+		},
+		ID:        laoID,
+		Name:      name,
+		Creation:  getTime(),
```

Comment:
You might want to reuse the one declared on line 53.

Offtopic, but this very likely indicates we're not verifying if laoID is consistent with `SHA256(organizer||creation||name)`

Reply:
I agree. I think we're currently not verifying if any ID is consistent with its hash described in the protocol (expect for the roll call ids). Should I open an issue for that?

Reply:
Yes, please do create one for it